crates/oxc_minifier/src/ast_passes/peephole_substitute_alternate_syntax.rs

```diff
@@ -503,6 +504,29 @@ impl<'a, 'b> PeepholeSubstituteAlternateSyntax {
         Some(ctx.ast.move_expression(&mut expr.right))
     }
 
+    /// `a || (b || c);` -> `(a || b) || c;`
+    fn try_rotate_logical_expression(
+        expr: &mut LogicalExpression<'a>,
+        ctx: Ctx<'a, 'b>,
+    ) -> Option<Expression<'a>> {
+        let Expression::LogicalExpression(right) = &mut expr.right else { return None };
+        if right.operator != expr.operator {
+            return None;
+        }
+        let new_expr = ctx.ast.expression_logical(
+            expr.span,
+            ctx.ast.expression_logical(
+                expr.span,
+                ctx.ast.move_expression(&mut expr.left),
+                expr.operator,
+                ctx.ast.move_expression(&mut right.left),
+            ),
+            expr.operator,
+            ctx.ast.move_expression(&mut right.right),
+        );
+        Some(new_expr)
+    }
```

Comment:
causes an infinite loop somewhere 😞

Reply:
technically not an infinite loop, this function needs to bulk change rather than doing 1x change per pass

Reply:
😅 minification is hard. this should be fixed